server/src/main/java/org/opensearch/search/aggregations/bucket/BestBucketsDeferringCollector.java

```diff
@@ -161,6 +162,7 @@ public void preCollection() throws IOException {
 
     @Override
     public void postCollection() throws IOException {
+        assert searchContext.searcher().getLeafContexts().isEmpty() || finished != true;
```

Comment:
Why we need first check here ?

Reply:
Was curios as well, comes from constructor: `this.finished = context.searcher().getLeafContexts().isEmpty();`